<launch>

  <arg name="rviz"     default="true" doc="whether to show Rviz" />
  <!-- <arg name="location"/> -->

  <arg name="map_yaml"                                                   doc="filepath to previously captured map (required)" />

  <!-- STRETCH DRIVER -->
  <!-- <param name="/stretch_driver/broadcast_odom_tf" type="bool" value="true"/>
  <param name="/stretch_driver/mode" type="string" value="navigation" />
  <include file="$(find stretch_core)/launch/stretch_driver.launch" pass_all_args="true"/> -->

  <!-- CENTERED BASE LINK -->
  <node name="centered_base_link_tf_publisher" pkg="tf" type="static_transform_publisher" args="-0.1 0 0 0 0 0 1 /base_link /centered_base_link 100" />

  <!-- LASER RANGE FINDER -->
  <!-- <include file="$(find stretch_core)/launch/rplidar.launch" /> -->

  <!-- MAP SERVER -->
  <node name="map_server" pkg="map_server" type="map_server" args="$(arg map_yaml)" />

  <!-- LOCALIZATION -->
  <include file="$(find amcl)/examples/amcl_diff.launch" />

  <!-- NAVIGATION -->
  <node pkg="move_base" type="move_base" respawn="false" name="move_base" output="screen">
    <param name="base_global_planner" value="waypoint_global_planner/WaypointGlobalPlanner"/>
    <rosparam file="$(find stretch_navigation)/config/common_costmap_params.yaml" command="load" ns="global_costmap" />
    <rosparam file="$(find stretch_navigation)/config/common_costmap_params.yaml" command="load" ns="local_costmap" />
    <rosparam file="$(find stretch_seeing_eye)/config/move_base/local_costmap_params.yaml" command="load" />
    <rosparam file="$(find stretch_seeing_eye)/config/move_base/global_costmap_params.yaml" command="load" />
    <rosparam file="$(find stretch_seeing_eye)/config/move_base/base_local_planner_params.yaml" command="load" />
    <remap from="/cmd_vel" to="/stretch/cmd_vel" />
  </node>

  <!-- VISUALIZE -->
  <node name="rviz" pkg="rviz" type="rviz" output="log" args="-d $(find stretch_navigation)/rviz/navigation.rviz" if="$(arg rviz)" />

</launch>
